Sources/MapFilter/ValueSliderView.swift

```diff
@@ -0,0 +1,321 @@
+
+        let convertedRect = valueSlider.convert(thumbRect, to: self)
+        return convertedRect
+    }
+}
+
+struct ValueSliderViewStyle {
+    static let trackColor: UIColor = .sardine
+    static let activeRangeTrackColor: UIColor = .primaryBlue
+    static let sliderThumbImage: UIImage? = UIImage(named: .sliderThumb)
+    static let activeSliderThumbImage: UIImage? = UIImage(named: .sliderThumbActive)
+    static let trackHeight: CGFloat = 3.0
+    static let activeRangeTrackHeight: CGFloat = 6.0
+}
+
+private extension ValueSliderView {
+    func setup() {
+        activeRangeTrackView.isHidden = true
+        addSubview(trackView)
+        addSubview(activeRangeTrackView)
+        addSubview(valueSlider)
+        addSubview(referenceValuesContainer)
+
+        let referenceViewsConstraints = setupReferenceValueView()
+
+        let activeRangeTrackViewLeadingAnchor = activeRangeTrackView.leadingAnchor.constraint(equalTo: trackView.leadingAnchor)
+
+        let activeRangeTrackViewTrailingAnchor = activeRangeTrackView.trailingAnchor.constraint(equalTo: trackView.trailingAnchor)
+        activeRangeTrackViewTrailingAnchor.identifier = activeRangeTrackViewTrailingAnchorIdentifier
+
+        NSLayoutConstraint.activate([
+            valueSlider.topAnchor.constraint(equalTo: topAnchor),
+            valueSlider.leadingAnchor.constraint(equalTo: leadingAnchor),
+            valueSlider.trailingAnchor.constraint(equalTo: trailingAnchor),
+
+            trackView.leadingAnchor.constraint(equalTo: valueSlider.leadingAnchor, constant: .verySmallSpacing),
+            trackView.trailingAnchor.constraint(equalTo: valueSlider.trailingAnchor, constant: -.verySmallSpacing),
+            trackView.centerYAnchor.constraint(equalTo: valueSlider.centerYAnchor),
+            trackView.heightAnchor.constraint(equalToConstant: ValueSliderViewStyle.trackHeight),
+            activeRangeTrackViewLeadingAnchor,
+            activeRangeTrackViewTrailingAnchor,
+            activeRangeTrackView.centerYAnchor.constraint(equalTo: valueSlider.centerYAnchor),
+            activeRangeTrackView.heightAnchor.constraint(equalToConstant: ValueSliderViewStyle.activeRangeTrackHeight),
+
+            referenceValuesContainer.topAnchor.constraint(equalTo: valueSlider.bottomAnchor, constant: .smallSpacing),
+            referenceValuesContainer.leadingAnchor.constraint(equalTo: valueSlider.leadingAnchor),
+            referenceValuesContainer.trailingAnchor.constraint(equalTo: valueSlider.trailingAnchor),
+            referenceValuesContainer.bottomAnchor.constraint(equalTo: bottomAnchor),
+        ] + referenceViewsConstraints)
+
+        accessibilityElements = [valueSlider]
+        updateAccesibilityValues()
+    }
+
+    func setupReferenceValueView() -> [NSLayoutConstraint] {
+        var constraints = [NSLayoutConstraint]()
+        referenceValueViews = range.enumerated().compactMap { (index, element) -> SliderReferenceValueView<ValueKind>? in
```

Comment:
Switch this around? 

`let view = referenceValueIndexes.map { index in`
`      let element = range[index].element`
`      return SliderReference....``
`}`

Reply:
👍